pkg/start/start.go

```diff
@@ -244,48 +243,42 @@ func (o *Options) getOpenShiftVersion() string {
 }
 
 func (o *Options) processInitialFeatureGate(ctx context.Context, configInformerFactory configinformers.SharedInformerFactory) (configv1.FeatureSet, featuregates.CvoGates, error) {
+	var startingFeatureSet configv1.FeatureSet
+	var cvoGates featuregates.CvoGates
+
 	featureGates := configInformerFactory.Config().V1().FeatureGates().Lister()
 	configInformerFactory.Start(ctx.Done())
-	configInformerFactory.WaitForCacheSync(ctx.Done())
+
+	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
```

Comment:
Just want to understand your reasoning on the number:
`30s` here are `WaitForCacheSync` (maybe `5s`?) + poll timeout (`25s`)?

Reply:
Well the whole functionality is "get the feature gate from the cluster". Previously we had 30s-ish timeout for the `Get` operation to succeed (previously the client was not informer-based, it was a get directly to apiserver), either through client-go internal retries or the previous poll timeout.

Now when the client is informer-based, the operation that can fail / timeout is the cache sync, not the get, so we apply the same timeout on the cache sync like we previously did for the get.